### PR TITLE
Match some BG update modules

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: src/game/stage/camera.c [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+    { 0x20, 0x20, 0x02, 0x1D }, //
+    { 0x20, 0x20, 0x02, 0x1C }, //
+    { 0x20, 0x20, 0x02, 0x1D }, //
+    { 0x20, 0x20, 0x02, 0x1D }, //
+    { 0x20, 0x40, 0x02, 0x1C }, //
+    { 0x20, 0x20, 0x02, 0x1D }, //
+    { 0x40, 0x20, 0x02, 0x1C }, //
+};
+
+void sub_801C068(u32 level)
```
Comment (freshollie) [inline src/game/stage/camera.c:310]: CreateCamera?

Comment (freshollie) [reply src/game/stage/camera.c:310]: Or maybe `InitCamera`?

File: src/game/stage/camera.c [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+    if (sStageInitProcedures[level] != NULL) {
+        sStageInitProcedures[level]();
+    }
+}
+
+void sub_801C708(s32, s32);
+
+#define BOSS_CAM_FRAME_DELTA_PIXELS 5
+
+void sub_801C394(void)
```
Comment (freshollie) [inline src/game/stage/camera.c:448]: CameraMain?

File: sym_iwram.txt [hunk 3]
```diff
@@ -2,6 +2,7 @@
 build/sa2/src/lib/m4a.o(.bss.code);
 build/sa2/src/multi_boot.o(.bss);
 /* space 0x20 */
+. = 0x00000408; gUnknown_03000408 = .; . += 2;
```
Comment (freshollie) [inline sym_iwram.txt:5]: Are you sure there are globals here? All game related globals should start after core

Comment (freshollie) [reply sym_iwram.txt:5]: This looks more to me like an iwram offset so probably a task member at offset 0x408

Comment (freshollie) [reply sym_iwram.txt:5]: IMO https://decomp.me/scratch/UPpy9

Comment (JaceCear) [reply sym_iwram.txt:5]: I don't know, it doesn'T make sense for me right now... but also the code doesn't match yet, so maybe?

File: src/game/stage/camera.c (hunk 1 above)
Comment (JaceCear) [reply src/game/stage/camera.c:310]: Yeah, I thought that could make sense as well but didn't just wanna do it without being 100% sure...

File: src/game/stage/camera.c (hunk 2 above)
Comment (JaceCear) [reply src/game/stage/camera.c:448]: It's not a task, but UpdateCamera should suffice, I think?

File: sym_iwram.txt (hunk 3 above)
Comment (freshollie) [reply sym_iwram.txt:5]: It's very unlikely to be referencing a global var which relates to anything there. IMO this is probably a mistake with a null pointer? Or at least a pointer to `IWRAM_START` instead

Comment (freshollie) [reply sym_iwram.txt:5]: I would definitely say remove the global until we are sure what's going on. It's not even referenced anywhere else so not really any reason to believe it's absolutely a global

Comment (JaceCear) [reply sym_iwram.txt]: Alright!
Really wondering what this is about!

File: src/game/stage/background/zone_3.c [hunk 4]
```diff
@@ -144,13 +149,15 @@ void StageBgUpdateZone3Acts12(s32 a, s32 b)
 
     if (IS_SINGLE_PLAYER) {
         if ((gPlayer.moveState & MOVESTATE_8000000) && (gUnknown_030054F4 >= 7)) {
-            if (gUnknown_03000408 == 0) {
-                gUnknown_03000408 = a;
+            Unk3000408 *unk = IWRAM_PTR(NULL);
```
Comment (freshollie) [inline src/game/stage/background/zone_3.c:152]: Nice, if this matches it seems more legit. IDK

**Review by freshollie — APPROVED** [review]

File: src/game/stage/background/zone_3.c (hunk 4 above)
Comment (JaceCear) [reply src/game/stage/background/zone_3.c:152]: Well it doesn't rn, but I'm not planning to spend more time on this specific one for now! :D

**Pull request merged**